src/fparser/two/C99Preprocessor.py

```diff
@@ -0,0 +1,561 @@
+                Cpp_Undef_Stmt, Cpp_Line_Stmt, Cpp_Error_Stmt,
+                Cpp_Warning_Stmt, Cpp_Null_Stmt)
+    for cls in cls_list:
+        obj = cls(reader)
+        if obj:
+            return obj
+    return obj
+
+#
+# ISO/IEC 9899: 1999 (C99)
+#
+
+#
+# 6.10 Preprocessing directives
+#
+
+
+#
+# Below construct for if-elif-else-endif blocks of preprocessor macros works
+# as long as they are within a common block of the Fortran code. However,
+# since preprocessor macros can potentially span, e.g., specification part and
+# execution part, this does not comply to the AST format of fparser. Hence,
+# this is commented out and if-stmts etc are treated as one-line pp macros
+# without being correlated to their counterparts.
+#
+
+# class Program(Program_2003):
+#
+#     # Add Preprocess if-constructs to the Program class by extending
+#     # the specification
+#     subclass_names = Program_2003.subclass_names[:]
+#     subclass_names.append('Cpp_If_Construct')
+
+
+# class Cpp_If_Construct(BlockBase):  # 6.10.1 Conditional inclusion
+#     '''
+#     C99 6.10.1 Conditional inclusion
+#
+#     <cpp-if-construct> = <cpp-if-stmt>
+#                             [ <block> ]
+#                          [ <cpp-elif-stmt>
+#                             [ <block> ]
+#                          ]...
+#                          [ <cpp-else-stmt>
+#                             [ <block> ]
+#                          ]
+#                          <cpp-endif-stmt>
+#     '''
+#
+#     subclass_names = []
+#     use_names = ['Cpp_If_Stmt', 'Block', 'Cpp_Elif_Stmt', 'Cpp_Else']
+#
+#     @staticmethod
+#     def match(string):
+#         return BlockBase.match(
+#             Cpp_If_Stmt, [Execution_Part_Construct,
+#                           Cpp_Elif_Stmt,
+#                           Execution_Part_Construct,
+#                           Cpp_Else_Stmt,
+#                           Execution_Part_Construct],
+#             Cpp_Endif_Stmt, string,
+#             enable_cpp_if_construct_hook=True)
+#
+#     def tofortran(self, tab='', isfix=None):
+#         tmp = []
+#         start = self.content[0]
+#         end = self.content[-1]
+#         tmp.append(start.tofortran(tab='', isfix=isfix))
+#         for item in self.content[1:-1]:
+#             if isinstance(item, (Cpp_Elif_Stmt, Cpp_Else_Stmt)):
+#                 tmp.append(item.tofortran(tab='', isfix=isfix))
+#             else:
+#                 tmp.append(item.tofortran(tab=tab+'  ', isfix=isfix))
+#         tmp.append(end.tofortran(tab='', isfix=isfix))
+#         return '\n'.join(tmp)
+
+
+class Cpp_If_Stmt(Base):
+    '''
+    C99 6.10.1 Conditional inclusion
+
+    if-stmt is  # if constant-expression new-line
+                | ifdef identifier new-line
+                | ifndef identifier new-line
+
+    '''
+
+    subclass_names = []
+    use_names = ['Cpp_Macro_Identifier']
+
+    _regex = re.compile(r"#\s*(ifdef|ifndef|if)\b")  # 'if' last, order matters
```

Comment:
This comment was actually a leftover from @martin-schlipf's work. Order would matter if word ending (via `\b`) wasn't explicitly required after the group of alternatives, since then `if` would also match `ifdef`. As this is not the case, I removed the comment.